jython/InitAllTurnoutsSensors.py

```diff
@@ -0,0 +1,170 @@
+#       )
+#
+#   In this case "known" means turnouts and sensors that are physically on the layout
+#   and also are described in one or more Panel files that have been loaded into the running
+#   JMRI program, usually the PanelPro flavor.
+#
+#######    The layout must be turned on and the Panel loaded 
+#######         before activating this script!!!!
+#
+#######    WE HAVE LEARNED THE HARD WAY, THAT PANEL FILES THAT ACTIATE Turnouts
+#######    AT LOAD TIME, RISK THE CONSEQUENCES OF MOVING A Turnout UNDER A TRAIN
+
+
+# Original script, circa 2014, was loosely based on several example scripts supplied in the distribution folder
+#   program:jython
+# by various authors.  Update suggestions are from Dave Sand, Bob Jacobsen, and others.
+# Most recent update conforms with PanelPro version 5.3.6
+
+# Author: Cliff Anderson, Copyright 2023
+# Part of the JMRI distribution
+
+
+import jmri
+import java
+from org.slf4j import LoggerFactory;
+
+# Inherit and override methods from AbstractAutomaton parent class
+class InitializeTurnoutsSensors(jmri.jmrit.automat.AbstractAutomaton):
+
+    # Get reference to the Logger
+    log = LoggerFactory.getLogger("jmri.jmrit.jython.exec.script.InitializeTurnoutsSensors");
+    #    All messages written to self.log.info( "text" ) are
+    #        visible in the JMRI System console window at runtime and 
+    #        are appended to file: preference:systemfiles\messages.log
+    #        are stored in file: preference:systemfiles\session.log
+    #   To output the debug messages, see instructions regarding "default.lcf" file at
+    #       https://www.jmri.org/help/en/html/doc/Technical/Logging.shtml
+    #   and append either of the following lines to the end of your copy of that file:
+    '''
+log4j.category.jmri.jmrit.jython.exec=DEBUG
+log4j.category.jmri.jmrit.jython.exec=TRACE
+    '''
+
+    # override AbstractAutomaton parent class initialization
+    def init(self):
+        # Probably redundant self protection
+        powermanager.setPower(jmri.PowerManager.ON)
+        # Pause to allow track power restoration, if it was off
+        self.waitMsec ( 500 )   # Inherit from AbstractAutomaton
+
+        self.closeTurnouts()
+        self.deacivateSensors()
+        return
+
+########    END OF InitializeTurnoutsSensors.init()
+
+    # override AbstractAutomaton parent class managed loop
+    def handle(self):
+        # Make a record for the log to indicate that we have completed the task
+        self.log.info( "All done" )
+        return False    # To run at most once, and die
+
+########    END OF InitializeTurnoutsSensors.handle()
+
+    # initialization of each physical Turnout in order of appearance on the system list
+    # Close ALL known physical layout turnouts but ignore the JMRI internal ones
+    def closeTurnouts(self) :
+
+        internalCounter = 0
+        physicalCounter = 0
+        self.log.info( "Looping through all known turnouts" )
```

Comment:
Minor comment:  Maybe Loop is a better word.

Reply:
Fixed.